agent/proxycfg/state.go

```diff
@@ -621,53 +617,54 @@ func (s *state) run() {
 			// etc on future updates.
 			snapCopy, err := snap.Clone()
 			if err != nil {
-				logger.Error("Failed to copy config snapshot for proxy",
-					"proxy-id", s.proxyID.String(), "error", err,
+				s.logger.Error("Failed to copy config snapshot for proxy",
+					"error", err,
 				)
 				continue
 			}
 
 			select {
 			// try to send
 			case s.snapCh <- *snapCopy:
-				logger.Trace("Delivered new snapshot to proxy config watchers",
-					"proxy-id", s.proxyID.String(),
-				)
+				s.logger.Trace("Delivered new snapshot to proxy config watchers")
+
+				// Allow the next change to trigger a send
+				coalesceTimer = nil
 
-			// avoid blocking if a snapshot is already buffered
+				// Skip rest of loop - there is nothing to send since nothing changed on
+				// this iteration
+				continue
+
+			// avoid blocking if a snapshot is already buffered, but queue up a retry with a timer
 			default:
-				logger.Trace("Failed to deliver new snapshot to proxy config watchers",
-					"proxy-id", s.proxyID.String(),
-				)
-			}
+				s.logger.Trace("Failed to deliver new snapshot to proxy config watchers")
 
-			// Allow the next change to trigger a send
-			coalesceTimer = nil
+				if coalesceTimer == nil {
```

Comment:
I think this looks right now. Do you think it's worth commenting this block of code since it wasn't obvious to us why it was needed before the PR so it could help future us remember why it's here?